SSA/Projects/LLVMRiscV/simpproc.lean

```diff
@@ -0,0 +1,144 @@
+import SSA.Projects.LLVMRiscV.PeepholeRefine
+import SSA.Projects.LLVMRiscV.LLVMAndRiscv
+import SSA.Projects.InstCombine.Tactic
+import SSA.Projects.RISCV64.PrettyEDSL
+import SSA.Projects.InstCombine.LLVM.PrettyEDSL
+import Lean
+
+open LLVMRiscV
+open RV64Semantics
+open InstCombine(LLVM)
+/- This file contains simplification procedures needed for the hybrid dialect.-/
+
+/-
+Disabled due to simproc implementation not being re-evaluated correctly
+on Lean version "4.20.0-nightly-2025-04-21" -/
+set_option Elab.async false
+
+@[simp_denote]
+private theorem valuation_var_last_eq.lemma {Ty : Type} [TyDenote Ty] {Γ : Ctxt Ty} {t : Ty}
+  {s : Γ.Valuation} {x : TyDenote.toType t} : (s.snoc x) (Ctxt.Var.last Γ t) = x := by
+  rfl
+
+open Lean Meta Elab in
+simproc [simp_denote] valuation_var_last_eq ((Ctxt.Valuation.snoc _ _) (Ctxt.Var.last _ _)) := fun e => do
+  let (_f, xs) := e.getAppFnArgs
+  let ty := xs[0]!
+  let s := xs[4]!
+  let x := xs[xs.size - 1 - 2]!
+  let proof ← mkAppOptM ``valuation_var_last_eq.lemma #[.some ty, .none, .none, .none, .some s, .some x]
+  return .visit {
+    expr := x,
+    proof? := proof
+  }
+
+theorem riscVArgsFromHybrid_nil_eq : riscvArgsFromHybrid (HVector.nil) = HVector.nil := rfl
+
+open Lean Meta Elab in
+/-- Convert a `List Expr` into an `Expr` by building calls to `List.nil` and `List.cons`.
+Note that the `ToExpr` instance of `List` is insufficient, since it perform a *deep expression cast*,
+where it converts any `List α` where `[ToExpr α]` into a `List Expr`. So, when given a list of
+expressions, like [.const Nat], instead of building `[Nat]`, it builds `[Lean.Expr.Const ``Nat]`
+(i.e.., it seralizes the `Expr` as well!).
+Instead, we want a shallow serialization, where we just build `List.cons (.const Nat) List.nil`.
+-/
+def listExprToExprShallow (type : Option Expr) : List Expr → MetaM Expr
+| .nil => mkAppOptM ``List.nil #[type]
+| .cons x xs => do
+  let tailExpr ← listExprToExprShallow type xs
+  mkAppOptM ``List.cons #[type, x, tailExpr]
+
```

Comment:
This feels like it'd ought to exist in the standard library already. A loogle search for the signature shows [`Lean.Meta.mkListLit`](https://leanprover-community.github.io/mathlib4_docs/Lean/Meta/AppBuilder.html#Lean.Meta.mkListLit), whose [implementation](https://github.com/leanprover/lean4/blob/81b85203c90439fb7d7e0cec32130d62d8cb7a2e/src/Lean/Meta/AppBuilder.lean#L517-L524) looks like it does the right thing. Can you check whether replacing listExprToShallow with Lean.Meta.mkListLit (and dropping the former) also works?